File: LeetCode/1.两数之和.java
import java.util.HashMap;
import java.util.Stack;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<Integer,Integer>();
        int [] ans = new int[2];
        for(int i = 0;i < nums.length;i ++) {
            if (map.get(target-nums[i]) != null) {
                ans[0] = i;
                ans[1] = map.get(target-nums[i]);
                break;
            }
            map.put(nums[i], i);
        }
        for(int i = 0;i < nums.length;i ++) {
            if (map.get(target-nums[i]) != null && map.get(target-nums[i])!=i) {
                ans[0] = i;
                ans[1] = map.get(target-nums[i]);
                break;
            }
        }
        return ans;
    }
}
// @lc code=end

File: LeetCode/1. 两数之和.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nums.sort()
        # print(nums)

        i, j = 0, len(nums)

        while i < j:
            if nums[i] + nums[j] < target:
                i += 1
            elif nums[i] + nums[j] == target:
                return [i, j]
            else:
                j -= 1